tc/sgx/trusted_worker_manager/enclave/kme/workload/kme_workload_plug-in.cpp

```diff
@@ -389,3 +677,64 @@ int KMEWorkloadProcessor::isSgxSimulator() {
 #endif  // defined(SGX_SIMULATOR)
 }
 
+std::string KMEWorkloadProcessor::serializeSigKeyMap(){
+    std::string serialized_str;
+    std::map<ByteArray, ByteArray>::iterator it;
+
+    for (it = sig_key_map.begin(); it != sig_key_map.end(); it++){
+        serialized_str += ByteArrayToStr(it->first);
+        serialized_str += ",";
+        serialized_str += ByteArrayToStr(it->second);
+        serialized_str += " ";
+    }
+
+    return serialized_str;
+}
+std::map<ByteArray, ByteArray> KMEWorkloadProcessor::derializeSigKeyMap(std::string serialized){
+    std::map<ByteArray, ByteArray> _sig_key_map;
+
+    while(serialized.length() != 0){
+        std::string::size_type pos = serialized.find_first_of(' ');
+        std::string map_entry = serialized.substr(0, pos);
+
+        std::string::size_type pos_entry = serialized.find_first_of(',');
+        std::string key = map_entry.substr(0, pos_entry);
+        std::string value = map_entry.substr(pos_entry+1);
+        _sig_key_map[StrToByteArray(key)] = StrToByteArray(value);
+
+        serialized = serialized.substr(pos+1);
+    }
+
+    return _sig_key_map;
+}
+std::string KMEWorkloadProcessor::serializeEncKeyMap(){
+    std::string serialized_str;
+    std::map<ByteArray, WPEInfo>::iterator it;
+
+    for (it = wpe_enc_key_map.begin(); it != wpe_enc_key_map.end(); it++){
+        serialized_str += ByteArrayToStr(it->first);
+        serialized_str += ":";
+        serialized_str += ByteArrayToStr((it->second).serialize());
+        serialized_str += "|";
```

Comment:
Why can't we use the same delimiter for serializing both encryrption and signature key maps.

Reply:
The WPEInfo causes the difference which also needs some delimiter.

Reply:
Ok